stripe/api_resources/search_result_object.py

```diff
@@ -1,10 +1,14 @@
 from __future__ import absolute_import, division, print_function
+from typing_extensions import Self
 
 from stripe.stripe_object import StripeObject
 
 
 class SearchResultObject(StripeObject):
     OBJECT_NAME = "search_result"
+    data: list
```

Comment:
`List[StripeObject]`?

Reply:
Let's give it a shot!

Reply:
I did that. That produced a type error when doing `data[0].id` below. What I really want is

```python
T = TypeVar("T", bound=Intersection[StripeObject, _HasId])
class SearchResultObject(StripeObject, Generic[T]):
   data: List[T]
```

but so far python doesn't have intersection types: (see https://github.com/python/typing/issues/213). So I just enforce the presence of id dynamically, not statically.